pkg/generate/app/cmd/template.go

```diff
@@ -38,9 +38,11 @@ func TransformTemplate(tpl *templateapi.Template, client client.TemplateConfigsN
 
 	// ensure the template objects are decoded
 	// TODO: in the future, this should be more automatic
-	if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
-		err = errors.NewAggregate(errs)
-		return nil, fmt.Errorf("error processing template %s: %v", name, err)
+	if decode {
+		if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
```

Comment:
In current master, `oc new-app` calls `runtime.DecodeList` on the list of objects that is returned by the API endpoint `/oapi/v1/namespaces/{namespace}/processedtemplates`, `oc process` does not call this function.

When I modify `oc process` to call this function on the list of objects received from server and then run `oc process -f test/testdata/template_required_params.yaml -v required_param=foo` it gives me the first snippet which looks wrong.

I guess my question is what does this function do? It doesn't seem to only validate the objects since it overwrites the items in the slice passed to it as the first argument.